### PR TITLE
Identity v2&v3: add RequestOpts.OmitHeaders, remove blank header

### LINKED ISSUE #2314: OpenStack V2 Authentication Failed since RequestOpts.MoreHeaders does not make sense
In `provider_client.go:322`, RequestOpts.Header says it will be omitted if it is a blank value. 
```
// RequestOpts customizes the behavior of the provider.Request() method.
type RequestOpts struct {
	// MoreHeaders specifies additional HTTP headers to be provide on the request. If a header is
	// provided with a blank value (""), that header will be *omitted* instead: use this to suppress
	// the default Accept header or an inferred Content-Type, for example.
	MoreHeaders map[string]string
}
```

But in commit 29581a2fc7076bae722d4b681f9cfaa467aa3e09 [Swift V1: fix setting empty headers (#2218)], the method locate in  ​`provider_client.go:411` has been changed, and blank header no longer been removed.

This caused an `Authentication Failed` error when trying to get an v2auth client due to an empty `X-Auth-Token` header.
This symptom is imported after v0.21.0.

`openstack/identity/v2/tokens/requests.go:94`:
```
resp, err := client.Post(CreateURL(client), b, &r.Body, &gophercloud.RequestOpts{
	OkCodes:     []int{200, 203},
	MoreHeaders: map[string]string{"X-Auth-Token": ""},
})
```

How can I make it correct? Is there any suggestion or work around?

### PR DESCRIPTION
**Pull request by freedywu**
Fixes #2314

### CODE REVIEW

File: provider_client.go [hunk 1]
```diff
@@ -412,6 +412,10 @@ func (client *ProviderClient) doRequest(method, url string, options *RequestOpts
 		}
 	}
 
+	for _, v := range options.OmitHeaders {
+		req.Header.Del(v)
```
Comment (pierreprinetti) [inline provider_client.go:416]: Is there any concern that the headers from `client.AuthenticatedHeaders()` will override this setting (line 420-422)?

Comment (freedywu) [reply provider_client.go:416]: Before commit 29581a2fc7076bae722d4b681f9cfaa467aa3e09, `client.AuthenticatedHeaders()` is prior to `options.MoreHeaders`. In my case, `options.OmitHeaders` makes up for the previous `options.MoreHeaders` and should inherit the original order

**Comment by freedywu** [comment]
@pierreprinetti

File: provider_client.go (hunk 1 above)
Comment (kayrus) [reply provider_client.go:419]: yeah, we expected that something could be broken, see https://github.com/gophercloud/gophercloud/issues/2215#issuecomment-921114591

Comment (jtopjian) [reply provider_client.go:419]: I'm not sure that something _is_ broken yet, though. I think there needs to be some investigation and comparison to @freedywu's use-case that is triggering this and the basic acceptance test that passes without changes required.

Comment (jtopjian) [reply provider_client.go:419]: > Is there any concern that the headers from client.AuthenticatedHeaders() will override this setting

I _think_ it should be okay given that the order of the existing code has `MoreHeaders` first adding an empty `X-Auth-Token` and then `client.AuthenticatedHeaders` adds headers on top of it. Definitely warrants running this PR through the Keystone suite of acceptance tests, though.

**Comment by jtopjian** [comment]
Per discussion in #2314, it looks like the root cause is a _very_ old version of Keystone triggering this issue. 

In my opinion, if the Keystone suite of acceptance tests passes with this PR, it should be fine to merge. It's not a lot of work to help get this functionality fixed and adds a feature that may be useful in other areas. And right now, the only place `OmitHeaders` is used is in the token creation functions, so that limits the scope of what's being modified.

**Comment by EmilienM** [comment]
@freedywu could you please rebase on master, so we get the new CI job for Keystone? Thanks

**Comment by freedywu** [comment]
> @freedywu could you please rebase on master, so we get the new CI job for Keystone? Thanks

Sure, thanks for noticing.

**Comment by freedywu** [comment]
Hi @EmilienM , my commit has been rebased, could you have a look at it?

**Review by mandre — APPROVED** [review]
LGTM, the code looks safe, and the identity acceptance tests worked as expected. 
Any objection to merge?

**Pull request merged**